src/common/keyboard.js

```diff
@@ -13,6 +13,94 @@ import * as OS from './os';
 
 const platform = OS.platform();
 
+const getCharCode =
+  ({charCode, keyCode}) =>
+  ( keyCode === 13
+  ? 13
+  : charCode == null
+  ? keyCode
+  : charCode >= 32
+  ? charCode
+  : 0
+  );
+
+
+if (!('key' in window.KeyboardEvent.prototype)) {
+  const keyTable =
+    { "3": "Enter"
+    , "8": "Backspace"
+    , "9": "Tab"
+    , "12": "Clear"
+    , "13": "Enter"
+    , "16": "Shift"
+    , "17": "Control"
+    , "18": "Alt"
+    , "19": "Pause"
+    , "20": "CapsLock"
+    , "27": "Escape"
+    , "32": "Space"
+    , "33": "PageUp"
+    , "34": "PageDown"
+    , "35": "End"
+    , "36": "Home"
+    , "37": "ArrowLeft"
+    , "38": "ArrowUp"
+    , "39": "ArrowRight"
+    , "40": "ArrowDown"
+    , "44": "PrintScreen"
+    , "45": "Insert"
+    , "46": "Delete"
+    , "112": "F1"
+    , "113": "F2"
+    , "114": "F3"
+    , "115": "F4"
+    , "116": "F5"
+    , "117": "F6"
+    , "118": "F7"
+    , "119": "F8"
+    , "120": "F9"
+    , "121": "F10"
+    , "122": "F11"
+    , "123": "F12"
+    , "144": "NumLock"
+    , "145": "ScrollLock"
+    , "224": "Meta"
+    , "91": "Meta"
+    , "92": "Meta"
+    , "93": "Meta"
+    , "63273": "Home"
+    , "63275": "End"
+    , "63276": "PageUp"
+    , "63277": "PageDown"
+    , "63302": "Insert"
+    };
+
+  const getKey = function() {
+    if (this.type === 'keypress') {
+      const charCode = getCharCode(this)
+      const key =
+        ( charCode === 13
+        ? keyTable[charCode]
+        : String.fromCharCode(charCode)
+        );
+
+      return key
+    }
+    else if (this.type === 'keyup' || this.type === 'keydown') {
+      return keyTable[this.keyCode]
```

Comment:
> We should never use keyCode. What is that for?

Webkit does not yet implement `KeyboardEvent.prototype.key` & this is a polyfill for Electron to handle keybindings, only defined if `KeyboardEvent.prototype.key` is not defined.